packages/agents/cartographer/poller/src/bindings/poller/index.ts

```diff
@@ -8,12 +8,31 @@ import { updateRouters } from "../../lib/operations/routers";
 // Ought to be configured properly for each network; we consult the chain config below.
 export const DEFAULT_SAFE_CONFIRMATIONS = 5;
 
-export const bindPoller = async (_pollInterval?: number) => {
+let stoppedExternally = false;
+
+const stopExternally = () => {
+  stoppedExternally = true;
+};
+
+process.on("SIGINT", function () {
+  stopExternally();
+});
+
+export const bindPoller = async (_pollInterval?: number, _pollIterations?: number) => {
```

Comment:
Should we just remove this completely and run a single iteration? I don't see the need to loop in JS anymore.

Reply:
We could, I thought so too. Looping within JS removes startup costs, and hence would perform slightly better, as it would autorestart after the last loop. And this would be best of both worlds.

Reply:
I have updated it to do a single iteration, and not loop in JS anymore.